lib/src/plural_resolver.dart

```diff
@@ -2,25 +2,38 @@ import 'dart:ui';
 
 import 'options.dart';
 
+/// A callback function to determine if exists
+/// the key for an specifc plural modifier
+typedef KeyExists = bool Function(
+  String pluralModifier,
+);
+
 class PluralResolver {
   PluralResolver() : super();
 
   /// Returns the plural suffix based on [count] and presented [options].
-  String pluralize(Locale locale, int count, I18NextOptions options) {
-    var result = '';
-    if (count != 1) {
-      final number = _numberForLocale(count.abs(), locale);
-      if (number >= 0) {
-        result = '${options.pluralSuffix}${options.pluralSeparator}$number';
-      } else {
-        result = '${options.pluralSeparator}${options.pluralSuffix}';
-      }
-    }
-    return result;
+  String pluralize(
+    int count,
+    I18NextOptions options,
+    KeyExists validateKey,
+  ) {
+    final shouldLookForPluralKeys = count != 1;
+    const resultForNonRequiredPlural = '';
+
+    return shouldLookForPluralKeys
+        ? _pluralKey(count, options, validateKey)
+        : resultForNonRequiredPlural;
```

Comment:
Something that came to mind now, with these changes there's the `count == 1` issue, right?

`key_1` is not allowed, since it is ignored by the early exit here.

Any thoughts on how to avoid that? Or to in some way mitigate the issue?

Reply:
:thinking: oh yeah, I missed this one, actually, I've decided this skipping a little about the i18next semantic

In my perspective, from one side, it should be good to keep the same semantic of i18next, it will be easy to learn how to use this package

Btw, on another side, it looks strange if the user defines this specific localization JSON

```json
{
  "clicked": "You clicked {{count}} time!",
  "clicked_1": "One click",
  "clicked_plural": "You clicked {{count}} times!"
}
```

Should we get the key with `_1` when count is equal 1?
What are ur thoughts?

Reply:
I changed the code allowing bigger priority for keys with suffix `_1`

```dart
    final isPlural = count != 1;
    final hasSpecificPlural = _hasSpecificPlural(count, options, validateKey);
    const resultForNonRequiredPlural = '';

    if (hasSpecificPlural) {
      return '${options.pluralSeparator}$count';
    }

    return isPlural ? '_plural' : resultForNonRequiredPlural;
``` 

If you agree with this approach I'll push the code for it

Reply:
The usage of multiple-plurals on i18next is only allowed to some locales (e.g. arabic), while others use the simple binary plural rule (`_plural`).

Maybe we should fix the special rule to known locales after all? (only the ones that require it)

Reply:
Yep, I think is a good idea, I need to search a little more about it, do you have this list of locales?

Reply:
no, but the official doc has it (or at least in their source code)